byteutils/byteutils.go

```diff
@@ -15,3 +15,18 @@ func ReadAvailableBytesToBuffer(target []byte, targetOffset int, source []byte,
 
 	return bytesToRead
 }
+
+// Concat concatenates the byte slices into a new byte slice.
+func Concat(byteSlices ...[]byte) (result []byte) {
```

Comment:
Is it really worth to have a until function for this as very similar functionality is provided by `bytes.buffer`?
The buffer even has advanced functionality like assuring that the cap is set correctly.